MetaMask/metamask-extension
# Subscribe multi

## **Description**



When using the per-dapp network selection feature flag, subscribe and filter requests are not respecting the per-dapp selection. This the main goal of this PR. Additionally, this PR correctly uses the persisted values for the feature flag. It also addresses an issue where changing the feature flag value would require dapps to be refreshed.  

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/6668 https://github.com/MetaMask/MetaMask-planning/issues/1870 https://github.com/MetaMask/MetaMask-planning/issues/1871 https://github.com/MetaMask/MetaMask-planning/issues/1865

## **Manual testing steps**

**note** to test subscribe:
  1. go to https://docs.metamask.io/wallet/reference/eth_subscribe/
  2. open dev tools / console and add event listener: `window.ethereum.on('message', ({data}) => console.log(parseInt(data.result.number, 16)));`
  3. hit execute on eth_subscribe
  4. console should now start showing block updates

  1. Basic Functionality Tests:
    - Verify that subscribing to different Ethereum events (like block headers, logs, or specific event types) works as expected.
    - Test filtering functionalities, ensuring that the filter correctly captures and displays events based on provided criteria.
  3. Network-Specific Tests:
    - Test the subscribe and filter functionalities across different networks (mainnet, Ropsten, Kovan, etc.) to ensure consistent behavior.
    - Check for any network-specific issues or inconsistencies in the functioning of the subscribe and filter features.
  5. Test with and without useRequestQueue Feature Flag:
    - Conduct tests with the useRequestQueue feature flag enabled and verify the functioning of the subscribe and filter features.
    - Repeat the tests with the feature flag disabled to ensure that the functionalities remain stable and consistent.
  6. Repeat the above tests, but in a multi-dapp scenario
  


## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
